domain/network/state/space.go

```diff
@@ -303,3 +303,43 @@ WHERE  space_uuid = $space.uuid;`, sp)
 	})
 	return err
 }
+
+// IsSpaceUsedInConstraints checks if the provided space name is used in any
+// constraints.
+// This method doesn't check if the provided space name exists, it returns
+// false in that case.
+func (st *State) IsSpaceUsedInConstraints(ctx context.Context, name network.SpaceName) (bool, error) {
+	db, err := st.DB()
+	if err != nil {
+		return false, errors.Capture(err)
+	}
+
+	sp := spaceName{Name: string(name)}
+	stmt, err := st.Prepare(`
+SELECT COUNT(*) AS &countResult.count
+FROM (
+	SELECT NULL AS n
+	FROM v_model_constraint_space
+	WHERE space = $spaceName.name
+	UNION 
+	SELECT NULL AS n
```

Comment:
`AS n` is superfluous right?

Reply:
It is, but we have a check on sqlfluff against ambiguous assignments so we need it.